packages/EasyCodingStandard/src/config/services.yml

```diff
@@ -16,3 +16,10 @@ services:
             # exact in that order, so Fixer can cleanup after Sniffer
             - ['addFileProcessor', ['@Symplify\EasyCodingStandard\SniffRunner\Application\SniffFileProcessor']]
             - ['addFileProcessor', ['@Symplify\EasyCodingStandard\FixerRunner\Application\FixerFileProcessor']]
+
+    Jean85\Version:
+        factory: ['Jean85\PrettyVersions', 'getVersion']
+        arguments: ['symplify/easy-coding-standard']
+
+    Symplify\EasyCodingStandard\Console\Application:
+        arguments: ['EasyCodingStandard', '@Jean85\Version']
```

Comment:
I didn't like it at first sight, but it respects DI approach better.

Just one WTF thing is that `@Jean85\Version` is not a services but a string. In some other case, provider or factory would be better, but here we need just the string.

Since it's not reusable anywhere, I'd make it clear like this:

```yml
Symplify\EasyCodingStandard\Console\Application:
    arguments: 
        - 'EasyCodingStandard'
        # not sure what the exact syntax is, but you get the idea
        - Jean85\PrettyVersions::getVersion('symplify/easy-coding-standard')
```

Reply:
@Jean85\Version is a service with `__toString` method, that's why it works. And suggested syntax is not possible.

Reply:
I like to have configuration in configuration file. In contrast to having it mixed in application code, it allows to wire application code with different configuration. In theory it allows to instantiate Application class with different parameters. This would not be possible if constructor override was hardcoded there like it was originally. Not saying somebody would want to do that in this case, but this is general advantage of DI as opposed to inheritance.

IDE autocomple works for me, but this is really small chunk of config, so not a big problem.

But hey, I don't care much. I can revert this to previous way if you want.

Reply:
How to make PHPStorm to autocomplete methods in config? 

Well it's not readable for me and programing in config usually leads to huge overcomplexity.
Mixing value object to magic __toString() is no go.

Please revert